libs/coin-modules/coin-tezos/src/logic/listOperations.ts

```diff
@@ -98,14 +123,25 @@ function convertOperation(
       },
       date: new Date(operation.timestamp),
     },
-    type: type,
+    type: normalizedType,
     value: amount,
     senders: senders,
     recipients: recipients,
     details: {
       counter: operation.counter,
       gasLimit: operation.gasLimit,
       storageLimit: operation.storageLimit,
+      status: hasFailed ? "failed" : operation.status,
     },
   };
+
+  if (isAPIDelegationType(operation)) {
+    coreOp.details!.ledgerOpType = operation.newDelegate?.address ? "DELEGATE" : "UNDELEGATE";
+  } else if (isAPIRevealType(operation)) {
+    coreOp.details!.ledgerOpType = "REVEAL";
+  } else if (normalizedType === "FEES") {
+    coreOp.details!.ledgerOpType = "FEES";
+  }
```

Comment:
Oh ok, I did not realize `!` was needed. Then you can probably extract the logic and do
```suggestion
      status: hasFailed ? "failed" : operation.status,
      ledgerOpType: getLedgerOpType(operation),
    },
  };
```